plugins/modules/ndo_l3out_interface_group_policy.py

```diff
@@ -0,0 +1,512 @@
+    type: str
+  state:
+    description:
+    - Determines the desired state of the resource.
+    - Use C(absent) to remove the resource.
+    - Use C(query) to list the resource.
+    - Use C(present) to create or update the resource.
+    type: str
+    choices: [ absent, query, present ]
+    default: query
+notes:
+- The O(template) must exist before using this module in your playbook.
+  Use M(cisco.mso.ndo_template) to create the L3Out template.
+- The O(l3out) must exist before using this module in your playbook.
+  Use M(cisco.mso.ndo_l3out_template) to create the L3Out object under the L3Out template.
+- The O(interface_routing_policy) must exist before using it in the module.
+  Use M(cisco.mso.ndo_l3out_interface_routing_policy) to create the L3Out Node Routing Policy.
+- The O(custom_qos_policy) must exist before using it in the module.
+  Use M(cisco.mso.ndo_tenant_custom_qos_policy) to create the Custom QoS Policy.
+seealso:
+- module: cisco.mso.ndo_template
+- module: cisco.mso.ndo_l3out_template
+- module: cisco.mso.ndo_l3out_interface_routing_policy
+- module: cisco.mso.ndo_tenant_custom_qos_policy
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Create an L3Out interface group policy
+  cisco.mso.ndo_l3out_interface_group_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: l3out_template
+    l3out: l3out
+    name: interface_group_policy_1
+    description: Test description
+    interface_routing_policy:
+      name: ansible_l3out_interface_routing_policy
+      template: ansible_tenant_template
+    bfd:
+      enable: true
+      authentication: true
+      key_id: 1
+      key: TestKey
+    bfd_multi_hop:
+      enable: true
+      authentication: true
+      key_id: 1
+      key: TestKey
+    ospf:
+      enable: true
+      authentication_type: simple
+      key_id: 1
+      key: TestKey
+    custom_qos_policy:
+      template: ansible_tenant_template
+      name: ansible_custom_qos_policy
+    qos_priority: level1
+    state: present
+
+- name: Query an existing L3Out interface group policy with name
+  cisco.mso.ndo_l3out_interface_group_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: l3out_template
+    l3out: l3out
+    name: interface_group_policy_1
+    state: query
+  register: query_with_name
+
+- name: Query all existing L3Out interface group policies under l3out
+  cisco.mso.ndo_l3out_interface_group_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: l3out_template
+    l3out: l3out
+    state: query
+
+- name: Delete an existing L3Out interface group policy with name
+  cisco.mso.ndo_l3out_interface_group_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: l3out_template
+    l3out: l3out
+    name: interface_group_policy_1
+    state: absent
+"""
```

Comment:
Examples for update, query and delete with UUID

Reply:
There's no UUID associated with this object